eventauth_test.go

```diff
@@ -1591,5 +1602,340 @@ func TestMembershipAllowed(t *testing.T) {
+					"state_key": "@u3:a",
+					"event_id": "$e2:a",
+					"content": {"membership": "knock"}
+				},
+				"@u4:a": {
+					"type": "m.room.member",
+					"sender": "@u2:a",
+					"room_id": "!r1:a",
+					"state_key": "@u4:a",
+					"event_id": "$e2:a",
+					"content": {"membership": "invite"}
+				}
+			}
+		},
+		"allowed": [{
+			"type": "m.room.member",
+			"sender": "@u2:a",
+			"room_id": "!r1:a",
+			"state_key": "@u1:a",
+			"event_id": "$e4:a",
+			"content": {"membership": "leave"}
+		}, {
+			"type": "m.room.member",
+			"sender": "@u2:a",
+			"room_id": "!r1:a",
+			"state_key": "@u3:a",
+			"event_id": "$e4:a",
+			"content": {"membership": "ban"}
+		}, 
+		{
+			"type": "m.room.member",
+			"sender": "@u2:a",
+			"room_id": "!r1:a",
+			"state_key": "@u3:a",
+			"event_id": "$e4:a",
+			"content": {"membership": "ban"}
+		}, 
+		{
+			"type": "m.room.member",
+			"sender": "@u2:a",
+			"room_id": "!r1:a",
+			"state_key": "@u4:a",
+			"event_id": "$e4:a",
+			"content": {"membership": "ban"}
+		}
+		],
+		"not_allowed": [{
+			"type": "m.room.member",
+			"sender": "@u1:a",
+			"room_id": "!r1:a",
+			"state_key": "@u1:a",
+			"event_id": "$e4:a",
+			"content": {"membership": "join"},
+			"unsigned": {
+				"not_allowed": "Sender is not in room"
+			}
+		},
+		{
+			"type": "m.room.member",
+			"sender": "@u1:a",
+			"room_id": "!r1:a",
+			"state_key": "@u1:a",
+			"event_id": "$e4:a",
+			"content": {"membership": "knock"},
+			"unsigned": {
+				"not_allowed": "Sender is not in room"
+			}
+		},
+		{
+			"type": "m.room.member",
+			"sender": "@u2:a",
+			"room_id": "!r1:a",
+			"state_key": "@u1:a",
+			"event_id": "$e4:a",
+			"content": {"membership": "invite"},
+			"unsigned": {
+				"not_allowed": "Sender is not in room"
+			}
+		}]
+	}`, RoomVersionV10)
+}
+
+func TestJoinRuleInvite(t *testing.T) {
+	testEventAllowed(t, `{
+		"auth_events": {
+			"create": {
+				"type": "m.room.create",
+				"state_key": "",
+				"sender": "@u1:a",
+				"room_id": "!r1:a",
+				"event_id": "$e1:a",
+				"content": {"creator": "@u1:a"}
+			},
+			"join_rules": {
+				"type": "m.room.join_rules",
+				"state_key": "",
+				"sender": "@u1:a",
+				"room_id": "!r1:a",
+				"event_id": "$e1:a",
+				"content": {"join_rule": "invite" }
+			},
+			"member": {
+				"@u2:a": {
+					"type": "m.room.member",
+					"sender": "@u2:a",
+					"room_id": "!r1:a",
+					"state_key": "@u2:a",
+					"event_id": "$e2:a",
+					"content": {"membership": "invite"}
+				}
+			}
+		},
+		"allowed": [{
+			"type": "m.room.member",
+			"sender": "@u2:a",
+			"room_id": "!r1:a",
+			"state_key": "@u2:a",
+			"event_id": "$e2:a",
+			"content": {"membership": "join"}
+		}],
+		"not_allowed": [{
+			"type": "m.room.member",
+			"sender": "@u1:a",
+			"room_id": "!r1:a",
+			"state_key": "@u1:a",
+			"event_id": "$e2:a",
+			"content": {"membership": "join"},
+			"unsigned": {
+				"not_allowed": "Sender not invited or joined"
+			}
+		}]
+	}`, RoomVersionV1)
```

Comment:
Room version 10 I presume?

Reply:
Shouldn't matter for `join_rule` set to `invite`, there were no changes to this, if I saw that correctly.